desktop/libs/aws/src/aws/conf.py

```diff
@@ -352,9 +349,13 @@ def has_s3_access(user):
   from desktop.conf import RAZ  # Must be imported dynamically in order to have proper value
 
   return user.is_authenticated and user.is_active and (
-    is_admin(user) or user.has_hue_permission(action="s3_access", app="filebrowser") or \
-    (RAZ.IS_ENABLED.get() and bool(get_raz_s3_default_bucket()))
-  )
+    is_admin(user) or user.has_hue_permission(action="s3_access", app="filebrowser") or RAZ.IS_ENABLED.get())
+
+
+def has_raz_s3():
+  from desktop.conf import RAZ  # Must be imported dynamically in order to have proper value
+
+  return (RAZ.IS_ENABLED.get() and bool(get_raz_s3_default_bucket()))
```

Comment:
At some point, using has_raz_s3() above (instead of duplicated logic here and there) and adding one more check like below would prevent to show up the S3 Browser for ADLS (when RAZ is also on)

(
RAZ.IS_ENABLED.get() and (
'default' in list(AWS_ACCOUNTS.keys()) and AWS_ACCOUNTS['default'].get_raw() and AWS_ACCOUNTS['default'].HOST.get()
)